file/file_prefetch_buffer.cc

```diff
@@ -587,7 +650,7 @@ Status FilePrefetchBuffer::PrefetchInternal(const IOOptions& opts,
       // Data is overlapping i.e. some of the data has been copied to overlap
       // buffer and remaining will be updated below.
       size_t initial_buf_size = overlap_buf_->CurrentSize();
-      CopyDataToBuffer(buf, offset, length);
+      CopyDataToOverlapBuffer(buf, offset, length);
```

Comment:
Should this entire block (i.e starting at line 648) be skipped if `use_fs_buffer` is true? Right now, its a bit hard to reason about that case. We may have already copied to overlap buffer, and we'll attempt to copy again here, which is confusing.

Reply:
I agree that the logic is a bit hard to reason about. I have added an explanation as a comment (starting with `Note: why do we not end up performing a duplicate copy`. 

When `use_fs_buffer` is true and we have only 1 buffer, `buf->DoesBufferContainData()` will return false so we'll skip this block. When we have more than 1 buffer, we will have relevant data to be copied to the overlap buffer, since we always clean up "outdated" buffers.

Reply:
To prove this is correct, I updated this PR to also have a `FilePrefetchBuffer::CopyDataToOverlapBuffer:Complete` sync point which I use in the tests to count the number of times we copy data to the overlap buffer. This helped me confirm that the overlap buffer copying logic worked as I had expected it to.